pkg/relay/relay.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package relay
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+)
+
+type Relay struct {
+	addr          *net.UDPAddr
+	bufferChannel chan string
+	logger        log.Logger
+	packetLength  uint
+}
+
+// NewRelay creates a statsd UDP relay. It can be used to send copies of statsd raw
+// lines to a separate service.
+func NewRelay(l log.Logger, target string, packetLength uint) (*Relay, error) {
+	addr, err := net.ResolveUDPAddr("udp", target)
+	if err != nil {
+		return nil, fmt.Errorf("unable to resolve target %s, err: %w", target, err)
+	}
+
+	c := make(chan string, packetLength)
```

Comment:
I was thinking this was the largest size line that would be accepted over the channel.

Reply:
The chan doesn't care how long each string is, it effectively only handles the metadata struct. In make, you are saying how many strings the channel can hold before it blocks.

Reply:
I don't know what a good queue would be. Having 1400 lines in queue seems a bit much, switching goroutines is not _that_ expensive. Let's set it to 100 as a start?

```suggestion
	c := make(chan string, 100)
```

Reply:
Ohh, my confusion. Yea, 100 things buffered in the channel sounds fine.

Reply:
Based on a recommendation from a coworker, I changed this to a `[]byte` channel and a `bytes.Buffer()`. Rather than using string manipulation in the buffer handling.